api/scpca_portal/test/models/test_job.py

```diff
@@ -6,62 +6,96 @@
 
         job.submit()
-
-        mock_batch_client.submit_job.assert_called_once_with(
-            jobName=self.batch_job_name,
-            jobQueue=settings.AWS_BATCH_JOB_QUEUE_NAME,
-            jobDefinition=settings.AWS_BATCH_JOB_DEFINITION_NAME,
-            containerOverrides=self.project_job_batch_container_overrides,
-        )
+        mock_batch_client.submit_job.assert_called_once()
 
         # After submission, the job should be updated
-        self.assertEqual(job.batch_job_id, self.mock_job_id)
+        self.assertEqual(job.batch_job_id, self.mock_batch_job_id)
         self.assertEqual(job.state, JobStates.SUBMITTED)
         self.assertIsInstance(job.submitted_at, datetime)
 
         # Make sure that the job is saved in the db with correct field values
         self.assertEqual(Job.objects.count(), 1)
         saved_job = Job.objects.first()
-        self.assertEqual(saved_job.batch_job_name, self.batch_job_name)
+        self.assertEqual(saved_job.batch_job_id, self.mock_batch_job_id)
         self.assertEqual(saved_job.batch_job_queue, settings.AWS_BATCH_JOB_QUEUE_NAME)
         self.assertEqual(saved_job.batch_job_definition, settings.AWS_BATCH_JOB_DEFINITION_NAME)
-        self.assertEqual(
-            saved_job.batch_container_overrides, self.project_job_batch_container_overrides
-        )
-        self.assertIn(self.project_id, saved_job.batch_container_overrides["command"])
+        self.assertIn("--project-id", saved_job.batch_container_overrides["command"])
+        self.assertIn(self.mock_project_id, saved_job.batch_container_overrides["command"])
         self.assertEqual(saved_job.state, JobStates.SUBMITTED)
         self.assertIsInstance(saved_job.submitted_at, datetime)
+
+    @patch("scpca_portal.models.Job._batch")
+    def test_terminate_job(self, mock_batch_client):
+        # Set up mock for the SUBMITTED job
+        submitted_job = JobFactory(
+            batch_job_name=self.mock_project_batch_job_name,
+            batch_job_id=self.mock_batch_job_id,
+            state=JobStates.SUBMITTED,
+        )
+
+        success = submitted_job.terminate(retry_on_termination=True)
+        mock_batch_client.terminate_job.assert_called()
+        self.assertTrue(success)
+
+        # After termination, the job should be saved with correct field values
+        saved_job = Job.objects.first()
+        self.assertEqual(saved_job.state, JobStates.TERMINATED)
+        self.assertTrue(saved_job.retry_on_termination)
+        self.assertIsInstance(saved_job.terminated_at, datetime)
+
+        # Set up mock for the TERMINATED job
+        terminated_job = JobFactory(
+            batch_job_name=self.mock_project_batch_job_name,
+            batch_job_id=self.mock_batch_job_id,
+            state=JobStates.TERMINATED,
+        )
+
+        # Should return Ture for previously terminated job
+        success = terminated_job.terminate(retry_on_termination=True)
+        mock_batch_client.terminate_job.assert_called()
```

Comment:
based on my read this should fail so lets take a closer look

Reply:
I separated the tests for the submitted jobs (`test_terminate_job`) and terminated jobs (`test_terminate_job_on_terminated_job`) to prevent conflicting assertions.

Reply:
Alternatively, instead of separating the tests, we can assert that the mock method wasn't called twice for the terminated job by using the following assertion:
- `self.assertFalse(mock_batch_client.terminate_job.call_count, 2)` (this may be a better approach?) 
- `mock_batch_client.terminate_job.assert_called_once()` (make sure called only once within the test)

Let me know your thoughts on this.

Reply:
Alternatively, instead of separating the tests, we can assert that the mock method wasn't called twice for the terminated job by using one of the following assertions:
- `self.assertEqual(mock_batch_client.terminate_job.call_count, 1)` (this may be a better approach?) 
- `mock_batch_client.terminate_job.assert_called_once()`  

Let me know your thoughts on this.

Reply:
I've improved readability by swapping the order of assertions for `TERMINATED` and `SUBMITTED` jobs, and replaced `call_count` with `assert_not_called` for the `TERMINATED` job in d7089e2. Thank you!